### DISCUSSION TITLE
Allow reverse proxy via base URL/subfolder

### DISCUSSION THREAD

**Discussion by ghost** [umami-software/umami#858] [Q&A]
I'm hosting Umami and am able to successfully configure a reverse proxy via analytics.mydomain.com, but cannot figure out a valid configuration (via Nginx) to serve Umami at app.mydomain.com/analytics.

Is this something that would ever be considered for a future release?

**Comment by mikecao** [comment]
Use BASE_PATH, https://umami.is/docs/environment-variables

**Comment by ghost** [comment]
> Use BASE_PATH, https://umami.is/docs/environment-variables

Is there anything I need to do w/ my proxy conf once I've updated the base path? I've added it and it looks like the page isn't rendering any HTML - all I see is a giant umami logo when I navigate to app.mydomain.com/analytics.

Here's what I'm currently using:

```
location /analytics {
	
        include /config/nginx/proxy.conf;
        include /config/nginx/resolver.conf;
        set $upstream_app umami;
        set $upstream_port 3000;
        set $upstream_proto http;
        proxy_pass $upstream_proto://$upstream_app:$upstream_port;
		
}
```

Edit: It might be worth asking - should the "BASE_PATH" variable be configured as "/analytics" or "app.mydomain.com/analytics"?

**Comment by mikecao** [comment]
Should be `/analytics`. As for nginx I'm not sure, You'll have to dive through stackoverflow to find the right settings.

**Comment by ghost** [comment]
> Should be `/analytics`. As for nginx I'm not sure, You'll have to dive through stackoverflow to find the right settings.

I'll definitely continue exploring different options, but for what it's worth - I use several other applications that support a base URL and use the same Nginx config for them all. I wonder what's different about your implementation that requires a more precise configuration.

**Comment by mikecao** [comment]
The basePath feature is from Next.js, https://nextjs.org/docs/api-reference/next.config.js/basepath

**Comment by agarsev** [comment]
It seems that umami (or Next.js) expects the url to keep the base path after proxying.

With nginx, if you use proxy_pass it removes the base path. For umami, don't remove it, eg

```
location /umami {
    ...
    proxy_pass http://127.0.0.1:3000/umami;
    ...
}
```

**Reply by ghost** [reply]
I've tried this and now instead of returning nothing, I'm brought to a page with a giant Umami logo saying "Page not found". It seems we're getting closer and closer to getting this to work!

**Comment by dodedodo** [comment]
Hey! 

I'm having the exact same problem. 

For the sake of troubleshooting, I've spun up umami in docker without any reverse proxy (like nginx). The `BASE_PATH` environmental variable simply does not work. Whatever I put in, umami only listens to `/`.

I think this issue was already raised in #557, but for some reason it's closed.

**Comment by dtcooper** [comment]
Also running into this issue. `BASE_PATH` doesn't seem to be respected when passing it to the umami docker container via an environment variable, @mikecao

**Comment by dannyking** [comment]
I'm running into this issue too. When I try to set up Umami to be at /umami using nginx I get the giant Umami logo saying "Page not found" too.

**Comment by tstanciu** [comment]
~~Same problem here. BASEPATH doesn't seem to work either with the docker image or with the local build :(~~

BASE_PATH works:
- Locally must be run every time "yarn build" + "yarn start" after modifying the ".env" file.
- For docker, the image must be built with the BASE_PATH ENV provided during the build. I think it is needed at the build time, and at runtime it becomes useless.
`docker image build --build-arg BASE_PATH=/umami-analytics -t "umami-app:1.0.1" .`

**Comment by mqu** [comment]
You may have a look at this [issue](https://github.com/umami-software/umami/issues/557#issuecomment-2751219773) to build new Docker image with BASE_PATH=/some-value
